src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1358,6 +1370,13 @@ func (k *kataAgent) createContainer(ctx context.Context, sandbox *Sandbox, c *Co
 
 	ctrStorages = append(ctrStorages, epheStorages...)
 
+	k.Logger().WithField("ociSpec Hugepage Resources", ociSpec.Linux.Resources.HugepageLimits).Debug("ociSpec HugepageLimit")
+	hugepages, err := k.handleHugepages(ociSpec.Mounts, ociSpec.Linux.Resources.HugepageLimits)
```

Comment:
Is `/sys/kernel/mm/hugepages/` per container? This seems to set a per container hugepage limit. What happens if two containers both set it?

Reply:
Now, this PR is a directly porting of the original PRs, modifying of `/sys/kernel/mm/hugepages/` seems like over-writing but not accumulating, so I think the second container's resource limitation will overwrite the first one.

Also both 2M and 1G in one Pod are not tested in these PRs(including PR in tests repo).